src/tests/psyGen_test.py

```diff
@@ -810,3 +815,794 @@ def test_invalid_reprod_pad_size():
             "REPROD_PAD_SIZE in config.py should be a positive "
             "integer") in str(excinfo.value)
     config.REPROD_PAD_SIZE = keep
+
+
+def test_argument_dependent_arg():
+    '''Check that the dependent_arg method returns the appropriate boolean
+    value for arguments with combinations of read and write access'''
+    _, invoke_info = parse(os.path.join(BASE_PATH,
+                                        "4.5_multikernel_invokes.f90"),
+                           distributed_memory=False, api="dynamo0.3")
+    psy = PSyFactory("dynamo0.3", distributed_memory=False).create(invoke_info)
+    invoke = psy.invokes.invoke_list[0]
+    schedule = invoke.schedule
+    arg_f1_inc_1 = schedule.children[0].children[0].arguments.args[0]
+    arg_f1_inc_2 = schedule.children[2].children[0].arguments.args[0]
+    arg_f2_read_1 = schedule.children[0].children[0].arguments.args[2]
+    arg_f2_inc = schedule.children[1].children[0].arguments.args[0]
+    arg_f2_read_2 = schedule.children[2].children[0].arguments.args[1]
+    # different names returns False
+    assert not arg_f2_inc._dependent_arg(arg_f1_inc_1)
+    # same name both reads returns False
+    assert not arg_f2_read_1._dependent_arg(arg_f2_read_2)
```

Comment:
_dependent_arg just looks at two argument objects and specifies whether there is a dependence between them or not based on their name and their access. Different names do not depend on each other and if both accesses are read then they do not depend on each other. The logic here is quite simple and is what all the more complex methods are built on. Could you clarify what you think needs changing as it looks OK to me (but of course I'm naturally blinkered about my own code).